webhook/webhook.go

```diff
@@ -136,6 +136,43 @@ type GenericCRD interface {
 	runtime.Object
 }
 
+// NewAdmissionController constructs an AdmissionController
+func NewAdmissionController(
+	client kubernetes.Interface,
+	opts ControllerOptions,
+	handlers map[schema.GroupVersionKind]GenericCRD,
+	logger *zap.SugaredLogger,
+	ctx func(context.Context) context.Context,
+	disallowUnknownFields bool) *AdmissionController {
+
+	reporter, err := NewStatsReporter()
+	if err != nil {
+		panic(err)
```

Comment:
Don't Panic (c). Can we log fatal, perhaps?

Reply:
Or return `*AdmissionController, error`